### PR TITLE
[OTE_SDK] CVS-78747 rename .id to .id_

### PR DESCRIPTION
**Pull request by Harmjandevries**
As discussed previously we will be renaming the attribute .id to .id_.

Active e2e: https://ci.iotg.sclab.intel.com/job/IMPT/job/SC-MR-dev/1585/

### CODE REVIEW

**Comment by nervana-ff** [comment]
Can one of the admins verify this patch?

**Comment by Harmjandevries** [comment]
run ote_sdk tests

**Comment by Harmjandevries** [comment]
run ote_sdk tests

**Comment by Harmjandevries** [comment]
Hi @Ilya-Krylov and @samet-akcay, these are the ID changes required for sonoma creek. I've not updated the "id" parameter in the init functions yet, this is something we will do in the future.

Running e2e: https://ci.iotg.sclab.intel.com/job/IMPT/job/SC-MR-dev/1540/

**Comment by samet-akcay** [comment]
@Harmjandevries would you be able to check this out, causing the fail in the CI

```
16:06:43  File [0;32m/home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/usecases/exportable_code/prediction_to_annotation_converter.py:178[0m, in [0;36mDetectionBoxToAnnotationConverter.convert_to_annotation[0;34m(self, predictions, metadata)[0m
16:06:43  [1;32m    176[0m image_size [38;5;241m=[39m metadata[[38;5;124m"[39m[38;5;124moriginal_shape[39m[38;5;124m"[39m][[38;5;241m1[39m::[38;5;241m-[39m[38;5;241m1[39m]
16:06:43  [1;32m    177[0m [38;5;28;01mfor[39;00m box [38;5;129;01min[39;00m predictions:
16:06:43  [0;32m--> 178[0m     scored_label [38;5;241m=[39m ScoredLabel([38;5;28mself[39m[38;5;241m.[39mlabels[[38;5;28mint[39m([43mbox[49m[38;5;241;43m.[39;49m[43mid_[49m)], [38;5;28mfloat[39m(box[38;5;241m.[39mscore))
16:06:43  [1;32m    179[0m     coords [38;5;241m=[39m np[38;5;241m.[39marray(box[38;5;241m.[39mget_coords(), dtype[38;5;241m=[39m[38;5;28mfloat[39m) [38;5;241m/[39m np[38;5;241m.[39mtile(image_size, [38;5;241m2[39m)
16:06:43  [1;32m    180[0m     annotations[38;5;241m.[39mappend(
16:06:43  [1;32m    181[0m         Annotation(
16:06:43  [1;32m    182[0m             Rectangle(coords[[38;5;241m0[39m], coords[[38;5;241m1[39m], coords[[38;5;241m2[39m], coords[[38;5;241m3[39m]),
16:06:43  [1;32m    183[0m             labels[38;5;241m=[39m[scored_label],
16:06:43  [1;32m    184[0m         )
16:06:43  [1;32m    185[0m     )
16:06:43  
16:06:43  [0;31mAttributeError[0m: 'Detection' object has no attribute 'id_'
```

**Comment by Harmjandevries** [comment]
run ote_sdk tests

**Comment by Harmjandevries** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
@Harmjandevries please resolve conflicts and what are the next steps?

**Comment by Harmjandevries** [comment]
> @Harmjandevries please resolve conflicts and what are the next steps?

Step 1 (this PR): Replace .id with .id_ in ote_sdk and add .id attribute so we don't break SC
Step 2: Replace .id with .id_ in sc_sdk
Step 3: Remove unused .id attribute from OTE

Conflicts are resolved, the only failed checks are unrelated to this MR. (test_ote_eval_openvino)

**Comment by samet-akcay** [comment]
run ote_sdk tests

**Comment by samet-akcay** [comment]
The changes themselves look alright to me

**Comment by Harmjandevries** [comment]
e2e passed except for known failures that are currently discussed in the validation channel. http://validationreports.sclab.intel.com:8004/reports/build_number_report?test_session_build_number=SC-MR-dev-1589&environment=idrefactore2e

I see a lot of tests are failing in the CI but these are related to the module 'torch' it seems, any suggestions for how to proceed?

**Comment by samet-akcay** [comment]
@Ilya-Krylov , is it possible to merge this one if this PR is relevant to those fails?

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/entities/scored_label.py [hunk 1]
```diff
@@ -31,11 +31,16 @@ def name(self) -> str:
...
+    def id_(self) -> ID:
         """
         Returns the label id.
         """
-        return self.label.id
+        return self.label.id_
+
+    @property
+    def id(self) -> ID:
+        """Gets or sets the id of a Model"""
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/entities/scored_label.py]: it does not set, I guess
why Model?

File: ote_sdk/ote_sdk/entities/resultset.py [hunk 2]
```diff
@@ -89,13 +89,22 @@ def __init__(
...
-        return self.__id
+        return self.__id_
+
+    @id_.setter
+    def id_(self, value: ID) -> None:
+        self.__id_ = value
+
+    @property
+    def id(self) -> ID:
+        """Gets or sets the id of a Model"""
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/entities/resultset.py]: it does not set, I guess
why `Model`?

File: ote_sdk/ote_sdk/entities/scored_label.py (hunk 1 above)
Comment (Ilya-Krylov) [reply ote_sdk/ote_sdk/entities/scored_label.py]: setter is required 
or
should it not be DEPRECATED?

File: ote_sdk/ote_sdk/entities/resultset.py (hunk 2 above)
Comment (Ilya-Krylov) [reply ote_sdk/ote_sdk/entities/resultset.py]: should it not be DEPRECATED?

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: ote_sdk/ote_sdk/entities/scored_label.py (hunk 1 above)
Comment (Harmjandevries) [reply ote_sdk/ote_sdk/entities/scored_label.py]: Thanks for finding that Ilya, I've corrected it.

File: ote_sdk/ote_sdk/entities/resultset.py (hunk 2 above)
Comment (Harmjandevries) [reply ote_sdk/ote_sdk/entities/resultset.py]: Corrected, thanks.

**Comment by Harmjandevries** [comment]
https://ci.iotg.sclab.intel.com/job/IMPT/job/SC-MR-dev/1626/

**Comment by Ilya-Krylov** [comment]
@Harmjandevries could you re-run build?

**Comment by Harmjandevries** [comment]
New build: https://ci.iotg.sclab.intel.com/job/IMPT/job/SC-MR-dev/1639/

**Comment by Ilya-Krylov** [comment]
BMM 25548 :clock1:

**Comment by Harmjandevries** [comment]
run ote_sdk tests

**Pull request merged**